pageserver/src/repository/rocksdb.rs

```diff
@@ -480,9 +488,53 @@ impl RocksTimeline {
         Ok(0)
     }
 
+    ///
+    /// Drop relations with all its forks or non-relational file
+    ///
+    fn drop(&self, tag: BufferTag) -> Result<()> {
+        let mut iter = self.db.raw_iterator();
+        let mut key = CacheKey {
+            tag,
+            lsn: Lsn(u64::MAX),
+        };
+        if tag.rel.forknum == pg_constants::MAIN_FORKNUM {
+            // if it is relation then remove all its blocks in all forks
+            key.tag.blknum = u32::MAX;
+            key.tag.rel.forknum = pg_constants::INIT_FORKNUM;
+        } else {
+            assert!(tag.rel.forknum > pg_constants::INIT_FORKNUM);
+        }
+        debug!("Drop relation {:?}", tag);
+        iter.seek_for_prev(key.ser()?);
+        while iter.valid() {
+            let key = CacheKey::des(iter.key().unwrap())?;
+            if key.tag.rel.relnode != tag.rel.relnode
+                || key.tag.rel.spcnode != tag.rel.spcnode
+                || key.tag.rel.dbnode != tag.rel.dbnode
+                || (key.tag.rel.forknum != tag.rel.forknum
+                    && tag.rel.forknum != pg_constants::MAIN_FORKNUM)
```

Comment:
`RelTag` impls `PartialEq + Eq`, so it looks like the first four lines can be replaced with `key.tag != tag.rel` ?

Reply:
Nope, the logic is more complex here than just lexicographical comparison of two structures.

Reply:
Then I suggest making this logic into a function in `RelTag`, so there's a place to document what this comparison means.

Reply:
Alternatively, we could wait and try turning forknum into an enum first; it might make this logic unnecessary?

Reply:
Another way to simplify places like that would be to add helper function fork(forknum: u8) to RelTag impl, that would return cloned RelTag with replaced forknum. Then this code could be:
```
if key.tag.rel != tag.rel
  || tag.rel.forknum != tag.rel.fork(pg_constants::MAIN_FORKNUM)
{
<...>
}
```

Reply:
Ah, nope. That is not what this check does.

Reply:
```
if !(key.tag.rel == tag.rel
  && tag.rel.forknum == tag.rel.fork(pg_constants::MAIN_FORKNUM)
```
?

Reply:
Once again wrong shot.
But fortunately this extra check is needed only in presence of non-relational data.
So I will just remove it in this branch.

Reply:
yeah, should be
```
if key.tag.rel != tag.rel
  && key.tag.rel != tag.rel.fork(pg_constants::MAIN_FORKNUM)
```